backend/posts/serializers.py

```diff
@@ -51,6 +51,8 @@ def validate(self, data):
         if computed_hash != data_hash:
             raise serializers.ValidationError('The data-hash is invalid.')
 
+        computed_hash = bytes.fromhex(computed_hash[2:])  # remove 0x
```

Comment:
Maybe just `computed_hash.lstrip('0x')`? Would be safer (applies if data is a string).